financial-connections/src/main/java/com/stripe/android/financialconnections/image/StripeImageLoader.kt

```diff
@@ -0,0 +1,131 @@
+package com.stripe.android.financialconnections.image
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.util.Log
+import java.io.InputStream
+import java.net.URL
+
+/**
+ * Image loader that fetches images from memory, disk or network and runs
+ * cache policy accordingly.
+ *
+ * [StripeImageLoader] is stateful as it holds the memoryCache instance. For
+ * memory cache to work the image loader instance needs to be shared.
+ */
+internal class StripeImageLoader(
+    context: Context
+) {
+    private var currentInputStream: InputStream? = null
+
+    private val diskCache = ImageLruDiskCache(
+        context = context,
+        uniqueName = "financial_connections_image_cache"
+    )
+    private val memoryCache = ImageLruMemoryCache()
+
+    fun load(
+        url: String,
+        width: Int,
+        height: Int
+    ): Result<Bitmap> {
+        return loadFromMemory(url)
+            ?: loadFromDisk(url)
+            ?: loadFromNetwork(url, width, height)
+                .onSuccess {
+                    debug("Image loaded from internet")
+                    diskCache.put(url, it)
```

Comment:
Instead of introducing another dep, could we buffer the downloaded file to `context.cacheDir` and use it's name as the key?

Reply:
[DiskLruCache](https://github.com/JakeWharton/DiskLruCache) basically does that, plus allows us to have an LRU policy and limit the size of our image cache. It's [pretty lightweight](https://github.com/JakeWharton/DiskLruCache/tree/master/src/main/java/com/jakewharton/disklrucache), and most image loading libraries implement disk cache this way. Thoughts?

Reply:
Ok, seems the [last release](https://search.maven.org/artifact/com.jakewharton/disklrucache/2.0.2/jar) of it is in 2013, I think it's pretty safe to use it